File: testdata/fixtures/scala/CWE-798.scala
// CWE-798: Hardcoded credentials — long secret written into source
object HardcodedTest {
  // ruleid: scala-hardcoded-credential
  val password = "SuperSecret123!"
  // ruleid: scala-hardcoded-credential
  val apiKey = "aBcD1234EfGh5678IjKlMnOpQrStUvWx"
  // ruleid: scala-hardcoded-credential
  var secret = "MyHardcodedSecretKey123"
  // ok: scala-hardcoded-credential
  val passwordPlaceholder = "changeme"
  // ok: scala-hardcoded-credential
  val apiKeyEnv = System.getenv("API_KEY")
  // ok: scala-hardcoded-credential
  val token = "test"
  // ok: scala-hardcoded-credential
  val short = "abc"
}

File: testdata/fixtures/rust/CWE-798.rs
// CWE-798: Hardcoded credentials in Rust
fn main() {
    // ruleid: rust-hardcoded-credential
    let password = "SuperSecret123!";
    // ruleid: rust-hardcoded-credential
    let api_key = "aBcD1234EfGh5678IjKlMnOpQrStUvWx";
    // ruleid: rust-hardcoded-credential
    let secret = "MyHardcodedSecretKey123";
    // ok: rust-hardcoded-credential
    let password_placeholder = "changeme";
    // ok: rust-hardcoded-credential
    let api_key_env = std::env::var("API_KEY").unwrap();
    // ok: rust-hardcoded-credential
    let token = "test";
    // ok: rust-hardcoded-credential
    let short = "abc";
}
